c++ training/SteamProject/Videogame/Category.cpp

```diff
@@ -0,0 +1,40 @@
+#include "Category.h"
+
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+FCategory::FCategory() : Name("Default Name"), CurrentNumberOfGames(0)
+{
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+FCategory::FCategory(std::string NewName) : Name(NewName), CurrentNumberOfGames(0)
+{
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FCategory::AddGame(FVideogame Videogame)
+{
+	// This adds the game in the current CurrentNumberOfGames position and after that increases its value adding 1
+	Games[CurrentNumberOfGames++] = Videogame;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bool FCategory::IsFull() const
+{
+	return CurrentNumberOfGames >= MaxGamesAmount;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+std::string FCategory::GetName() const
+{
+	return Name;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+FVideogame FCategory::GetGame(int Index) const
+{
+	if (Index >= 0 && Index < CurrentNumberOfGames)
+	{
+		return Games[Index];
+	}
+	return FVideogame();
```

Comment:
inside the if you are already returning an FVideoGame object because Games[Index] is an array of FVideoGames

Reply:
Yup, however we should always return something when dealing with a function that has a return type. That simply returns a new default object in case the condition on the if statement isn't reached.